accepted/00000-caasp-disable-skuba-update.md

```diff
@@ -0,0 +1,89 @@
+
+## Manual patch installs
+
+`skuba-update` can be called manually to invoke `zypper patch` and the subsequent maintenance actions (defined by CaaS Platform) issued on a node in case an installed patch requires a system reboot.
+For this reason, Uyuni/SUSE Manager must invoke `skuba-update` to install available patches on a system because of post-install actions already defined by `skuba-update`.
+`skuba-update` does not offer the possiblity of specifying which available patches should be installed: all available patches are going to be installed.
+Uyuni/SUSE Manager must directly call `skuba-update` and not its associated system service as the latter would just annotate the available updates and exit (as per Uyuni/SUSE Manager configuration of `skuba-update` service).
+`skuba-update` does not offer any API at the time of writing.
+There are some changes that Uyuni/SUSE Manager must introduce to emphasize that all patches are going to be installed.
+In the next sections we are going to describe these changes.
+
+### User interface: Software > Patches
+
+When showing a single CaaS Platform system, the checkboxes for every available patch shown in Software > Patches must be all selected and disabled. When the user selects "Apply Patches", a new Salt state that calls `skuba-update` has to be dispatched by SaltServerActionService (only if the system if a CaaS Platform system).
+The same applies when showing a group of systems in System Set Overview in regards to patching.
+
+### User interfaces: Patches > Patch list
+
+When selecting a patch in Patches > Patch list and showing all affected systems, if one of the systems is a CaaS Platform node then its associated checkbox must be disabled: it should not be possible to individually install a single patch on this system.
+
+### API
+
+The API for `ErrataManager.applyErrata` must be enriched to support a generic patch action to install all available patches. In the case of a CaaS Platform, the existing API calls (that require a list of errata IDs to be passed as argument) should fail in case an errata ID is passed, and resort to the new Salt state described above when no errata ID is passed.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+<!-- Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future? -->
+
+- Patch install is global via `skuba-update`: there is not the possibility of pin-pointing the install of a single patch but rather all available patches will be installed when calling `skuba-update`. This drawback can be mitigated by carefully promoting patches via [Content Lifecycle Management](https://documentation.suse.com/external-tree/en-us/suma/4.0/suse-manager/administration/content-lifecycle.html).
```

Comment:
We can improve this, `skuba-update` could take a list of patches (if empty, all get installed), and only apply those. This would help in not having to implement the special case outlined in `User interfaces: Patches > Patch list`.

Reply:
I think we should work with what we have right now - so far it is a limitation that must be outlined, but I appreciate your willingness to change it.
Should I introduce it in a "Future developments" section?

Reply:
I'm fine keeping as it is, no worries, we can always revisit later.